tests/functional/syntax/test_for_range.py

```diff
@@ -138,27 +161,35 @@ def bar():
         pass
     """,
         StructureException(
-            "For loop has invalid number of iterations (0), the value must be greater than zero"
+            "For loop has invalid number of iterations (0), the value must be greater than zero",
+            "0",
         ),
     ),
     (
         """
 @external
-def bar():
-    x:uint256 = 1
-    for i in range(0, x + 1):
+def repeat(n: uint256) -> uint256:
+    for i in range(n, 6):
         pass
+    return x
     """,
-        StateAccessViolation("Value must be a literal integer"),
+        StructureException("Second element must be the first element plus a literal value", "6"),
     ),
 ]
 
+for_code_regex = re.compile(r"for .+ in (.*):")
+fail_test_names = [
+    f"{i}: {for_code_regex.search(code).group(1)} raises {type(err).__name__}"
+    for i, (code, err) in enumerate(fail_list)
+]
+
 
-@pytest.mark.parametrize("bad_code,expected_error", fail_list)
+@pytest.mark.parametrize("bad_code,expected_error", fail_list, ids=fail_test_names)
 def test_range_fail(bad_code, expected_error):
     with pytest.raises(type(expected_error)) as exc_info:
         compiler.compile_code(bad_code)
     assert expected_error.message == exc_info.value.message
+    assert expected_error.args[1] == exc_info.value.args[1].node_source_code
```

Comment:
this pattern is a bit weird since the second argument to a `VyperException` should be an ast node. maybe just make the expected node_source_code the third item in the tuple?

Reply:
Yeah I decided to [split the message as well](https://github.com/vyperlang/vyper/pull/3679/commits/7284de2a560dfe74a0c4bf78dc1d831101a5f2ff), as the way the exception is constructed is different